### PR TITLE
fix: prevent slack messages for share posts

### PR DESCRIPTION
**Pull request by rebelchris**
Prevent slack messages for share posts.

Added a private flag for post and inherited from source.

Related:
https://github.com/dailydotdev/streams/pull/7

WT-982 #done

### CODE REVIEW

**Review by el — APPROVED** [review]

**Review by idoshamun — CHANGES_REQUESTED** [review]
This looks great! The only thing missing is a CDC that updates posts privacy when the source's privacy changes

**Comment by rebelchris** [comment]
@idoshamun Was wondering about that, how can that happen?
Didn't see any events for changing the privacy (something you do in DB only?)

**Comment by idoshamun** [comment]
@rebelchris in the future this will probably be supported and I'm afraid to forget about this mechanism

**Comment by rebelchris** [comment]
@idoshamun I added the source CDC change

File: .infra/workers.ts [hunk 1]
```diff
@@ -102,6 +102,10 @@ export const workers: Worker[] = [
     topic: 'api.v1.member-joined-source',
     subscription: 'api.squad-activation',
   },
+  {
+    topic: 'api.v1.source-privacy-updated',
+    subscription: 'source-privacy-updated-api',
```
Comment (idoshamun) [inline .infra/workers.ts:107]: Fix according to the new convention
```suggestion
    subscription: 'api.source-privacy-updated',
```

File: src/common/pubsub.ts [hunk 2]
```diff
@@ -198,6 +199,13 @@ export const notifySettingsUpdated = (
   settings: ChangeObject<Settings>,
 ): Promise<void> => publishEvent(log, settingsUpdatedTopic, settings);
 
+export const notifySourcePrivacyUpdated = (
+  log: EventLogger,
+  sourceId: string,
+  privacy: boolean,
+): Promise<void> =>
+  publishEvent(log, sourcePrivacyUpdatedTopic, { sourceId, privacy });
```
Comment (idoshamun) [inline src/common/pubsub.ts:207]: Should we consider publishing the complete source object?

File: src/workers/sourcePrivacyUpdated.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {
+          sourceId,
+          messageId: message.messageId,
+          err,
+        },
+        'failed to trigger updating source privacy',
+      );
+      if (err.name === 'QueryFailedError') {
+        return;
+      }
```
Comment (idoshamun) [inline src/workers/sourcePrivacyUpdated.ts:40]: This needs to be removed. Otherwise, it may be out of sync

**Review by idoshamun — COMMENTED** [review]
Some minor changes

**Review by idoshamun — APPROVED** [review]

**Pull request merged**